src/components/Tab/TabItem/TabItem.tsx

```diff
@@ -15,29 +16,35 @@ export const TabItem: React.FC<TabItemProps> = ({
     isSelected = false,
     onClick,
     ...props
-}) =>
-    React.createElement(
-        component,
-        {
-            className: clsx(
-                'moonstone-tabItem',
-                `moonstone-tabItem_${size}`,
-                {'moonstone-tabItem_noLabel': !label},
-                {'moonstone-tabItem_selected': isSelected},
-                {'moonstone-reverse': isReversed},
-                'flexRow_center',
-                'alignCenter',
-                className
-            ),
-            disabled: isDisabled,
-            onClick,
-            ...props
-        },
-        (
-            <>
-                {icon && <icon.type {...icon.props} className={clsx('moonstone-tabItem_icon', icon.props.className)} size={(size === 'big') ? 'default' : size}/>}
+}) => {
+    const ref = useRef(null);
+    return (
+        React.createElement(
+            component,
+            {
+                className: clsx(
+                    'moonstone-tabItem',
+                    `moonstone-tabItem_${size}`,
+                    {'moonstone-tabItem_noLabel': !label},
+                    {'moonstone-tabItem_selected': isSelected},
+                    {'moonstone-reverse': isReversed},
+                    'flexRow_center',
+                    'alignCenter',
+                    className
+                ),
+                ref: ref,
+                role: 'tab',
+                'aria-selected': isSelected,
+                disabled: isDisabled,
+                onClick,
+                ...props,
+                ...onArrowNavigation({ref: ref, direction: 'horizontal'})
```

Comment:
I think it's better to add `..props` at the end to easily override attributes returned by `onArrowNavigation`, otherwise people who use that component cannot set `tabIndex`.